### PR TITLE
BDP-1091: Decouple Housekeeping From Circus Train

### PR DESCRIPTION
**Pull request by courtsVII**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -175,20 +182,5 @@
       <version>1.3</version>
       <scope>test</scope>
     </dependency>
```
Comment (massdosage) [inline pom.xml]: Move the logback exclusion back here so it doesn't show up as a diff

File: src/main/java/com/hotels/housekeeping/audit/AuditRevision.java [hunk 2]
```diff
@@ -18,77 +18,16 @@
...
-      return false;
-    }
-    AuditRevision other = (AuditRevision) obj;
-    if (id != other.id) {
-      return false;
-    }
-    return true;
-  }
-}
+}
```
Comment (massdosage) [inline src/main/java/com/hotels/housekeeping/audit/AuditRevision.java]: new line

File: src/test/java/com/hotels/housekeeping/TestLegacyReplicaPath.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.housekeeping;
+
+import com.hotels.housekeeping.model.LegacyReplicaPath;
+
+public class TestLegacyReplicaPath implements LegacyReplicaPath {
```
Comment (massdosage) [inline src/test/java/com/hotels/housekeeping/TestLegacyReplicaPath.java]: Hopefully we won't need this class as you'll have the default implementation you can use (or just extend)

File: src/main/java/com/hotels/housekeeping/model/DefaultLegacyReplicaPath.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.hibernate.envers.Audited;
+
+@Entity
+// @Converter doesn't work with @Audited until https://hibernate.atlassian.net/browse/HHH-9042 is released
+@Audited
+@Table(schema = "housekeeping", name = "legacy_replica_path",
```
Comment (ddcprg) [inline src/main/java/com/hotels/housekeeping/model/DefaultLegacyReplicaPath.java]: the schema name has changed here, I've still not looked at CT but we shouldn't change it yet until the separation is completely done so we can define our own entity for CT. If this is not possible we must make clear in CHANGELOG.md that after this version is installed they must run the vacuum tool and ditch the old DB

File: src/main/java/com/hotels/housekeeping/model/EntityLegacyReplicaPath.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import javax.persistence.PrePersist;
+
+import org.hibernate.envers.Audited;
+import org.joda.time.Instant;
+
+import com.google.common.base.Objects;
+
+@MappedSuperclass
+@Audited
+public abstract class EntityLegacyReplicaPath implements LegacyReplicaPath {
```
Comment (ddcprg) [inline src/main/java/com/hotels/housekeeping/model/EntityLegacyReplicaPath.java]: ok, I see

File: src/main/java/com/hotels/housekeeping/model/DefaultLegacyReplicaPath.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package com.hotels.housekeeping.model;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+
+import org.hibernate.envers.Audited;
+
+@Entity
```
Comment (ddcprg) [inline src/main/java/com/hotels/housekeeping/model/DefaultLegacyReplicaPath.java]: This probably needs to be conditioned so it's only injected if no other `EntityLegacyReplicaPath`s are defined

File: src/main/java/com/hotels/housekeeping/model/DefaultLegacyReplicaPath.java (hunk 4 above)
Comment (massdosage) [reply src/main/java/com/hotels/housekeeping/model/DefaultLegacyReplicaPath.java]: This is a default implementation, you'll see in the CT PR that it extends this class and sets the table name to "circus_train" so it's still backwards compatible.

**Pull request merged**